yt_dlp/extractor/animeonegai.py

```diff
@@ -0,0 +1,103 @@
+    _GEO_COUNTRIES = ['MX']
+    _TESTS = [{
+        'url': 'https://www.animeonegai.com/es/watch/4Bpd6m6gGYVxdspKn?time=0&serie=true',
+        'md5': 'TODO: md5 sum of the first 10241 bytes of the video file (use --test)',
+        'info_dict': {
+            'id': '4Bpd6m6gGYVxdspKn',
+            'ext': 'mp4',
+            'title': 'Ep. 1: Eikichi Onizuka, 22 años: ¡Comienza la legendaria práctica docente del ex delincuente!',
+            'thumbnail': 'https://img-cdn.b-cdn.net/images/ayB0woggovEvoieXJ_1733001177.jpg',
+            'description': 'Eikichi Onizuka, de 22 años, fue un reconocido miembro de una pandilla de motociclistas durante su época de preparatoria. Ahora, su sueño de convertirse en profesor se hace realidad al comenzar su práctica docente. Espera con entusiasmo vivir un romance prohibido con una estudiante, pero le asignan una clase de jóvenes problemáticos. Allí conoce a la hermosa Nanako Mizuki, quien lo invita a su habitación después de clases. Sin embargo, pronto descubre que todo parece ser una trampa...',
+            'tags': ['Romance', 'Comedia', 'Drama', 'Serie'],
+            'series': 'GTO',
+            'episode_number': 1,
+        },
+        'skip': 'Login Required',
+    }]
+    _DEVICE_ID = str(uuid.uuid4())
+    _HEADERS = {
+        'Authorization': None,
+        'Content-Type': 'application/json',
+        'Host': 'api.animeonegai.com',
+    }
+
+    def _perform_login(self, username, password):
+        if (username.lower() == 'token' and password is not None):
+            self._HEADERS['Authorization'] = f'Bearer {password}'
+            self.report_login()
+        elif (username.lower() != 'token' and password is not None):
+            raise ExtractorError('Login type not supported', expected=True)
+
+    def _real_extract(self, url):
+        if (self._HEADERS['Authorization'] is None):
+            raise ExtractorError(
+                'Use "--username token" and "--password <ott_token>" to log in. The ott_token can be found in Local Storage using the Developer Tools, after login in a browser window to Anime Onegai site', expected=True)
+
+        language, video_id = self._match_valid_url(url).group('language', 'id')
+
+        self._initialize_geo_bypass({
+            'countries': self._GEO_COUNTRIES,
+        })
+
+        chapter_data = self._download_json(
+            f'https://api.animeonegai.com/v1/restricted/chapter/entry/{video_id}',
+            video_id, 'Downloading Chapter Data', headers=self._HEADERS, query={'lang': language})
+
+        if (chapter_data.get('ID') == 0):
+            raise ExtractorError('Chapter data not found', expected=True)
+
+        asset_id = int_or_none(chapter_data.get('asset_id'))
+        asset_data = self._download_json(
+            f'https://api.animeonegai.com/v1/restricted/asset/public/{asset_id}',
+            video_id, 'Downloading Asset Data', headers=self._HEADERS, query={'lang': language, 'cache': 'true'})
```

Comment:
imo. let's just inline those
```suggestion
        asset_data = self._download_json(
            f'https://api.animeonegai.com/v1/restricted/asset/public/{chapter_data["asset_id"]}',
            video_id, 'Downloading Asset Data', headers=self._HEADERS, query={'lang': language, 'cache': 'true'})
```